ui/apps/platform/src/Components/CompoundSearchFilter/types.ts

```diff
@@ -59,8 +59,41 @@ export function isOnSearchPayload(payload: OnSearchPayloadItem[]): payload is On
     return payload.length !== 0;
 }
 
-export type OnSearchPayloadItem = {
-    action: 'ADD' | 'REMOVE';
-    category: string;
+export type OnSearchPayloadItem =
+    | OnSearchPayloadItemAdd
+    | OnSearchPayloadItemDelete
+    | OnSearchPayloadItemRemove;
+
+export type OnSearchPayloadItemAdd =
+    | OnSearchPayloadItemAppendString
+    | OnSearchPayloadItemAppendToggle
+    | OnSearchPayloadItemAssignSingle;
+
+export type OnSearchPayloadItemAppendString = {
+    action: 'APPEND_STRING'; // inputType: autocomplete, and so on
+} & OnSearchPayloadItemWithValue;
+
+export type OnSearchPayloadItemAppendToggle = {
+    action: 'APPEND_TOGGLE'; // inputType: select
```

Comment:
I unfortunately do not have a good alternative, but `'APPEND_TOGGLE'` makes me pause each time I read it to consider what it means. My first instinct is that `APPEND` and `TOGGLE` are two disjoint actions, and the combination looks odd.